src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::{
+    env, fs,
+    io::{stderr, Write},
+    path::Path,
+    process,
+};
+
+use getopts::Options;
+use libc::{c_int, getchar, putchar};
+
+fn usage(prog: &str) -> ! {
+    let path = Path::new(prog);
+    let leaf = path
+        .file_name()
+        .map(|x| x.to_str().unwrap_or("ykbf"))
+        .unwrap_or("ykbf");
+    writeln!(&mut stderr(), "Usage: {} [-h] <file.bf>", leaf).ok();
+    process::exit(1)
+}
+
+fn compile(txt: &str) -> (String, Vec<usize>) {
+    // BF comments are anything that's not a command character, so scrub all non-command
+    // characters.
+    let mut out = String::with_capacity(txt.len());
+    for c in txt.chars() {
+        match c {
+            '>' | '<' | '+' | '-' | '.' | ',' | '[' | ']' => {
+                out.push(c);
+            }
+            _ => (),
+        }
+    }
+    out.shrink_to_fit();
+
+    // Pre-calculate the [...] jumps as a map from PC -> PC.
+    let mut map = Vec::with_capacity(out.len());
```

Comment:
I don't understand the allocation logic here. You make enough space for possibly every position, then truncate it to zero?

Reply:
Huh? `with_capacity` mallocs memory but doesn't fill it. `resize` (on the next line) then fills it.

Reply:
Ah, my bad, I misread.

How about:
```
let map = vec![0; out.len()];
```

Reply:
Good idea.